lib/src/cli.rs

```diff
@@ -439,7 +454,7 @@ async fn upgrade(opts: UpgradeOpts) -> Result<()> {
             }
         }
     } else {
-        let fetched = crate::deploy::pull(sysroot, imgref, opts.quiet).await?;
+        let fetched = crate::deploy::pull(sysroot, spec.backend, imgref, opts.quiet).await?;
```

Comment:
In the branch above here where we handle `--check`, it doesn't know how to store cached updates properly for the podman backend.  This in turn means we don't have a good way to represent that in `bootc status` (it will always just be `None` currently)

Reply:
We have a lot of choice for how to represent this; it strongly relates to the question of image GC though. I think today, podman and c/storage generally handle this by creating containers which hold references to images.

Reply:
This also feels like more reason re: [your previous comment](https://github.com/containers/bootc/pull/215#discussion_r1629415493) to switch to using skopeo instead of podman, since we could use `skopeo inspect` to only fetch the metadata in the `--check` case.  If I'm following correctly from [the way ostree does it](https://github.com/ostreedev/ostree-rs-ext/pull/537) today it's similar in that it's just saving the manifest/config.

Reply:
Yep